packages/beacon-node/src/chain/opPools/aggregatedAttestationPool.ts

```diff
@@ -227,48 +272,142 @@ export class AggregatedAttestationPool {
+      // should not happen
+      if (!attestationGroupByIndexByDataHash) {
+        throw Error(`No aggregated attestation pool for slot=${slot}`);
+      }
+
+      const epoch = computeEpochAtSlot(slot);
+      // validateAttestation condition: Attestation target epoch not in previous or current epoch
+      if (!(epoch === stateEpoch || epoch === statePrevEpoch)) {
+        continue; // Invalid attestations
+      }
+      // validateAttestation condition: Attestation slot not within inclusion window
+      if (!(slot + MIN_ATTESTATION_INCLUSION_DELAY <= stateSlot)) {
+        continue; // Invalid attestations
+      }
+
+      const slotDelta = stateSlot - slot;
+      // CommitteeIndex    0           1            2    ...   Consolidation
+      // Attestations    att00  ---   att10  ---  att20  ---   0 (att 00 10 20)
+      //                 att01  ---     -    ---  att21  ---   1 (att 01 __ 21)
+      //                   -    ---     -    ---  att22  ---   2 (att __ __ 22)
+      for (const attestationGroupByIndex of attestationGroupByIndexByDataHash.values()) {
+        // sameAttDataCons could be up to MAX_ATTESTATIONS_PER_GROUP_ELECTRA
+        const sameAttDataCons: AttestationsConsolidation[] = [];
+        for (const [committeeIndex, attestationGroup] of attestationGroupByIndex.entries()) {
+          const notSeenAttestingIndices = notSeenValidatorsFn(epoch, slot, committeeIndex);
+          if (notSeenAttestingIndices === null || notSeenAttestingIndices.size === 0) {
+            continue;
+          }
+
+          if (
+            slotCount > 2 &&
+            consolidations.length >= MAX_ATTESTATIONS_ELECTRA &&
+            notSeenAttestingIndices.size / slotDelta < minScore
+          ) {
+            // after 2 slots, there are a good chance that we have 2 * MAX_ATTESTATIONS_ELECTRA attestations and break the for loop early
+            // if not, we may have to scan all slots in the pool
+            // if we have enough attestations and the max possible score is lower than scores of `attestationsByScore`, we should skip
+            // otherwise it takes time to check attestation, add it and remove it later after the sort by score
+            continue;
+          }
+
+          if (!validateAttestationDataFn(attestationGroup.data)) {
+            continue;
+          }
+
+          // TODO: Is it necessary to validateAttestation for:
+          // - Attestation committee index not within current committee count
+          // - Attestation aggregation bits length does not match committee length
+          //
+          // These properties should not change after being validate in gossip
+          // IF they have to be validated, do it only with one attestation per group since same data
+          // The committeeCountPerSlot can be precomputed once per slot
+          // for (const [i, {attestation, notSeenAttesterCount}] of attestationGroup
```

Comment:
Is this line of comment necessary?

Reply:
let the comment be there for now, later cleanups can alwayss happen